www/js/pages/self-exclusion/self-exclusion.controller.js

```diff
@@ -9,17 +9,23 @@
 (function() {
     angular.module("binary.pages.self-exclusion.controllers").controller("SelfExclusionController", SelfExclusion);
 
-    SelfExclusion.$inject = ["$scope", "$translate", "alertService", "websocketService"];
+    SelfExclusion.$inject = ["$scope", "$translate", "alertService", "websocketService", "validationService", "appStateService"];
 
-    function SelfExclusion($scope, $translate, alertService, websocketService) {
+    function SelfExclusion($scope, $translate, alertService, websocketService,
+        validationService, appStateService) {
         const vm = this;
+        vm.validation = validationService;
         vm.today = new Date();
         vm.minDate = vm.today.toISOString().slice(0, 10);
         vm.minDateTime = vm.today.toISOString();
         vm.nextSixWeeks = new Date(vm.today.getTime() + 7 * 6 * 24 * 60 * 60 * 1000).toISOString().slice(0, 10);
         vm.nextSixMonths = new Date(vm.today.getTime() + 30 * 6 * 24 * 60 * 60 * 1000).toISOString().slice(0, 10);
         vm.disableUpdateButton = true;
         vm.isDataLoaded = false;
+        const currency = sessionStorage.getItem('currency') || 'USD';
+        const currencyConfig = appStateService.currenciesConfig || {};
+        vm.fractionalDigitis = currencyConfig && Object.keys(currencyConfig).length &&
```

Comment:
It's better to use `!_.isEmpty(currencyConfig)` instead of `currencyConfig && Object.keys(currencyConfig).length`

Reply:
@morteza-binary you're right, thanks, I changed that

Reply:
You can remove lines 25 to 28 and add the `fractionalDigitis` variable in `validationService` as the factory's public properties and use it here of directly in template file.

Reply:
@morteza-binary done